symfony/plugins/orangehrmAdminPlugin/client/jobCategory/views/JobCategoryList.vue

```diff
@@ -0,0 +1,175 @@
+<template>
+  <div class="orangehrm-background-container">
+    <div class="orangehrm-paper-container">
+      <div class="orangehrm-header-container">
+        <oxd-text tag="h6">Job Category List</oxd-text>
+        <div>
+          <oxd-button label="Add" displayType="secondary" @click="onClickAdd" />
+        </div>
+      </div>
+      <oxd-divider class="orangehrm-horizontal-margin" />
+      <div>
+        <div class="orangehrm-horizontal-padding orangehrm-vertical-padding">
+          <div v-if="checkedItems.length > 0">
+            <oxd-text tag="span">
+              {{ checkedItems.length }} Job Category Selected
+            </oxd-text>
+            <oxd-button
+              label="Delete Selected"
+              displayType="label-danger"
+              @click="onClickDeleteSelected"
+              class="orangehrm-horizontal-margin"
+            />
+          </div>
+          <oxd-text tag="span" v-else> {{ total }} Job Category Found</oxd-text>
+        </div>
+      </div>
+      <div class="orangehrm-container">
+        <oxd-card-table
+          ref="dTable"
+          :headers="headers"
+          :items="items"
+          :selectable="true"
+          v-model:selected="checkedItems"
+          rowDecorator="oxd-table-decorator-card"
+        />
+      </div>
+      <div class="orangehrm-bottom-container">
+        <oxd-pagination
+          v-if="showPaginator"
+          :length="pages"
+          v-model:current="currentPage"
+        />
+      </div>
+    </div>
+
+    <delete-confirmation ref="deleteDialog"></delete-confirmation>
+  </div>
+</template>
+
+<script>
+import DeleteConfirmationDialog from "../components/dialogs/DeleteConfirmationDialog";
+
+export default {
+  data() {
+    return {
+      headers: [
+        { name: "name", title: "Job Category", style: { "flex-basis": "80%" } },
+        {
+          name: "actions",
+          title: "Actions",
+          style: { "flex-shrink": 1 },
+          cellType: "oxd-table-cell-actions",
+          cellConfig: {
+            delete: {
+              onClick: this.onClickDelete,
+              component: "oxd-icon-button",
+              props: {
+                name: "trash",
+              },
+            },
+            edit: {
+              onClick: this.onClickEdit,
+              props: {
+                name: "pencil-fill",
+              },
+            },
+          },
+        },
+      ],
+      items: [],
+      total: 0,
+      pages: 1,
+      currentPage: 1,
+      pageSize: 5,
+      showPaginator: false,
+      editItem: null,
+      checkedItems: [],
+    };
+  },
+
+  components: {
+    "delete-confirmation": DeleteConfirmationDialog,
+  },
+
+  watch: {
+    currentPage() {
```

Comment:
will move pagination related data, logics into mixin or helper class. any idea?

Reply:
I checked this. even if we move most of logic to a mixin, we will often have to override those fields per component basis since header fields change.

Reply:
Implemented in `src/core/util/composable/usePaginate.ts`